### ISSUE TITLE
Layout/MultilineMethodArgumentLineBreaks reports false positive for rails model scopes with lambda

### ISSUE THREAD

**Issue by pduersteler** [rubocop/rubocop#7906]
Using rubocop in a rails project, I noticed that the Layout/MultilineMethodArgumentLineBreaks cop maybe reports a false positive for the case of using a lambda as a second argument.

    class MyModel < ApplicationRecord
      scope :my_scope, lambda { |argument|
        joins(:something).where(foo: { bar: argument })
      }
    end

I'm not sure if I am doing something wrong/against the styleguide here, but I could not figure it out. Before, I was using `->(argument) {}` syntax which yielded a Style/Lambda offense. The block argument also belong on the same line, otherwise another offense is displayed.

--------

## Expected behavior

My expectation is to either not get an offense on that line. If there is an offense, a fix should be available in the documentation.

## Actual behavior

An offense is displayed without any information about how to fix this.

## Steps to reproduce the problem

See code block in the issue description above.

## RuboCop version

```
$ bundle exec rubocop -V
0.82.0 (using Parser 2.7.1.1, running on ruby 2.6.5 x86_64-darwin19)
```

**Comment by hlascelles** [comment]
Agree this seems to be a false positive.

**Comment by tc-joegiralt** [comment]
this is driving me crazy as well. has anyone found a work around for scopes and other class level DSLs?

**Comment by Drenmi** [comment]
I'll try to have a look at this over the weekend.

**Comment by netrusov** [comment]
ran into the same problem while refactoring our application. I've monkey-patched (although I'm strongly against this approach) cop to skip method calls which have block type as a last argument. the whole point of this patch can be represented as:

```diff
diff --git a/lib/rubocop/cop/layout/multiline_method_argument_line_breaks.rb b/lib/rubocop/cop/layout/multiline_method_argument_line_breaks.rb
index d7ed853e3..ed93add45 100644
--- a/lib/rubocop/cop/layout/multiline_method_argument_line_breaks.rb
+++ b/lib/rubocop/cop/layout/multiline_method_argument_line_breaks.rb
@@ -40,6 +40,9 @@ module RuboCop
           # ...then each key/value pair is treated as a method 'argument'
           # when determining where line breaks should appear.
           last_arg = args.last
+
+          return if last_arg&.block_type? && last_arg.first_line == node.first_line
+
           args = args[0...-1] + last_arg.children if last_arg&.hash_type? && !last_arg&.braces?
 
           check_line_breaks(node, args)
```

perhaps a new configurable attribute needs to be introduced for such cases

**Comment by netrusov** [comment]
looks like it was already fixed in https://github.com/rubocop/rubocop/pull/10812

with this configuration the patch above is not needed anymore:
```yaml
Layout/MultilineMethodArgumentLineBreaks:
  Enabled: true
  AllowMultilineFinalElement: true
```
